### PR TITLE
Update Google Tag Manager package for initialisation flow

### PR DESCRIPTION
**Pull request by Manfies**
Google Tag Manger initialisation.
It was not possible prevent auto activate GTM. 
Now there are two new options in module settings: 

- `autoInitOnClientSide` - Send GTM event automatically. Also it is inserting GTM script and noscript elements, if there aren't presetted on server side.
- `presetScriptsOnServerSide` - It is inserting GTM script on server side. It can be blocked with value false.

I also updated README.md file for these new options.
This can be very useful for implementing GDPR and other user scenarios.

### CODE REVIEW

**Comment by edopenni** [comment]
That's perfect! So you can prevent for both the script to be loaded in the DOM via server and also prevent it from being downloaded in the browser?

**Comment by Manfies** [comment]
@edopenni If you set `autoInitOnClientSide` as `false` and `autoInitOnClientSide` as `false` value it will prevent for loading via server and client. After it you can just enable GTM with `this.$gtm.init()` or  `ctx.$gtm.init()` from your custom plugin.

**Comment by edopenni** [comment]
@Manfies just one last question: what happens if you only set autoInitOnClientSide as `false` and leave autoInitOnClientSide as true? Will that make the browser still download the GTM scripts?

**Comment by Manfies** [comment]
@edopenni Oh. I made mistake. `presetScriptsOnServerSide` as false. I updated comment.
Yeah.
If `autoInitOnClientSide` is `true` - scripts of GTM will be presetted on client side.
If `presetScriptsOnServerSide` is `true` - scripts of GTM will be presetted on server side.

**Comment by edopenni** [comment]
@Manfies ah, ok, so the correct way to do it if I wanna only download the GTM scripts on-demand is by setting `presetScriptsOnServerSide` and `autoInitOnClientSide` as `false`, right?
If so, then merge this baby! 😬

**Comment by Manfies** [comment]
@edopenni Yeah. Correct. And use `$gtm.init()` method for it.
But we are waiting for review from @pi0.

**Comment by edopenni** [comment]
I'd make a small note on the `pushEvent` function. If you try to push an event and the GTM haven't been enabled, you'll have an error. Maybe something like this:

```javascript
  pushEvent(obj) {
    try {
      if (!window || !window[this.options.layer]) {
        if (this.options.presetScriptsOnServerSide || this.options.autoInitOnClientSide) { 
          // will only error if we actually asked to load GTM and it didn't work
          throw new Error('missing GTM dataLayer')
        }
        return;
      }
```

File: packages/google-tag-manager/package.json [hunk 1]
```diff
@@ -7,5 +7,8 @@
   "publishConfig": {
     "access": "public"
   },
-  "main": "index.js"
+  "main": "index.js",
+  "dependencies": {
+    "lodash": "^4.17.15"
```
Comment (pi0) [inline packages/google-tag-manager/package.json]: Not a part of this PR but would you please change using lodash to [defu](https://github.com/jsless/defu)?

File: packages/google-tag-manager/index.js [hunk 2]
```diff
@@ -12,7 +12,9 @@ module.exports = async function nuxtTagManager(_options) {
     query: {},
     scriptURL: '//www.googletagmanager.com/gtm.js',
     noscriptURL: '//www.googletagmanager.com/ns.html',
-    env: {} // env is supported for backward compability and is alias of query
+    env: {}, // env is supported for backward compability and is alias of query
+    autoInitOnClientSide: true,
```
Comment (pi0) [inline packages/google-tag-manager/index.js]: I would simplyify it to:

```js
init: {
  client: true,
  server: true
}
```

File: packages/google-tag-manager/index.js [hunk 3]
```diff
@@ -44,26 +46,44 @@ module.exports = async function nuxtTagManager(_options) {
     .join('&')
 
   // sanitization before to avoid errors like "cannot push to undefined"
+  const headScriptId = 'google-tag-manager-script'
```
Comment (pi0) [inline packages/google-tag-manager/index.js]: Does it makes sense to make id configurable?

File: packages/google-tag-manager/plugin.js [hunk 4]
```diff
@@ -6,6 +6,30 @@ class GTM {
     this.options = options
   }
   init() {
+
+    // insert scripts on client side if it not preseted on server side
```
Comment (pi0) [inline packages/google-tag-manager/plugin.js]: /cc @pimlie Isn't there any easier method for vue-meta checking scripts existence?

File: packages/google-tag-manager/index.js [hunk 5]
```diff
@@ -44,27 +46,45 @@ module.exports = async function nuxtTagManager(_options) {
...
     src: (options.scriptURL || '//www.googletagmanager.com/gtm.js') + '?' + queryString,
+    id: headScriptId,
     async: true
-  })
+  }
 
-  // prepend google tag manager <noscript> fallback to <body>
-  this.options.head.noscript.push({
+  const gtmNoScript = {
     hid: 'gtm-noscript',
```
Comment (pi0) [inline packages/google-tag-manager/index.js]: Should we also use `bodyScriptId` for hid?

**Comment by Manfies** [comment]
Hi @pi0. Thanks for the review! I will update this soon.

**Comment by pi0** [comment]
Awesome @Manfies. I'm moving this module to a separate repo right now. Maybe better work in new repo :)

**Comment by Manfies** [comment]
@pi0 Yes. Ok. When there will be a new repo, I will create a fork and update the code for your review from this PR.

**Comment by pi0** [comment]
Please checkout v2: https://github.com/nuxt-community/gtm-module/releases/tag/v2.0.0 ✌️ 

All feedbacks and improvements more than welcome!

**Pull request merged**